client_configuration_test.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2023 Snowflake Computing Inc. All rights reserved.
+
+package gosnowflake
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"path"
+	"strings"
+	"testing"
+)
+
+type PositiveTestCase struct {
+	Name             string
+	FileName         string
+	FileContents     string
+	ExpectedLogLevel *string
+	ExpectedLogPath  *string
+}
+
+type NegativeTestCase struct {
+	Name                          string
+	FileName                      string
+	FileContents                  string
+	ExpectedErrorMessageToContain string
+}
+
+func TestThatParsesConfiguration(t *testing.T) {
+	// given
+	dir := CreateTempDirectory(t, "conf_parse_tests_")
+	testCases := []PositiveTestCase{
+		{
+			Name:     "TestWithLogLevelUpperCase",
+			FileName: "config.json",
+			FileContents: `{
+				"common": {
+					"log_level" : "INFO",
```

Comment:
Can we have another test looping over all log levels? So we can have something like:
```
for ... := range []{"INFO", "DEBUG"...
```

Reply:
done